modules/channel-twilio/src/backend/client.ts

```diff
@@ -0,0 +1,265 @@
+          options.push({
+            label: button.title,
+            value: button.payload,
+            type: 'postback'
+          })
+        } else if (button.type === 'say_something') {
+          options.push({
+            label: button.title,
+            value: button.text,
+            type: 'say_something'
+          })
+        }
+      }
+
+      const args = { mediaUrl: picture ? [picture] : undefined }
+      await this.sendOptions(event, body, args, options)
+    }
+  }
+
+  async sendChoices(event: sdk.IO.Event) {
+    const options: MessageOption[] = event.payload.quick_replies.map(x => ({
+      label: x.title,
+      value: x.payload,
+      type: 'quick_reply'
+    }))
+    await this.sendOptions(event, event.payload.text, {}, options)
+  }
+
+  async sendDropdown(event: sdk.IO.Event) {
+    const options: MessageOption[] = event.payload.options.map(x => ({
+      label: x.label,
+      value: x.value,
+      type: 'quick_reply'
+    }))
+    await this.sendOptions(event, event.payload.message, {}, options)
+  }
+
+  async sendOptions(event: sdk.IO.Event, text: string, args: any, options: MessageOption[]) {
+    let body = `${text}\n`
+    for (let i = 0; i < options.length; i++) {
+      const option = options[i]
+      body += `\n${i + 1}. ${option.label}`
+    }
+
+    await this.kvs.set(event.target, options)
+
+    await this.sendMessage(event, { ...args, body })
+  }
+
+  async sendMessage(event: sdk.IO.Event, args: any) {
+    await this.twilio.messages.create({
+      ...args,
+      provideFeedback: false,
+      from: event.threadId,
+      to: event.target
+    })
+  }
+}
+
+export async function setupMiddleware(bp: typeof sdk, clients: Clients) {
+  bp.events.registerMiddleware({
+    description:
+      'Sends out messages that targets platform = Twilio.' +
+      ' This middleware should be placed at the end as it swallows events once sent.',
+    direction: 'outgoing',
+    handler: outgoingHandler,
+    name: MIDDLEWARE_NAME,
+    order: 100
+  })
+
+  async function outgoingHandler(event: sdk.IO.Event, next: sdk.IO.MiddlewareNextCallback) {
+    if (event.channel !== 'twilio') {
+      return next()
+    }
+
+    const client: TwilioClient = clients[event.botId]
+    if (!client) {
+      return next()
+    }
+
+    return client.handleOutgoingEvent(event, next)
+  }
+}
+
+export async function setupRouter(bp: typeof sdk, clients: Clients, route: string): Promise<sdk.http.RouterExtension> {
+  const router = bp.http.createRouterForBot('channel-twilio', {
+    checkAuthentication: false
+  })
+
+  router.post(route, async (req, res) => {
+    const { botId } = req.params
+    const client = clients[botId]
+
+    if (!client) {
+      res.status(404).send('Bot not a twilio bot')
+    }
+
+    if (client.auth(req)) {
+      await client.handleWebhookRequest(req.body)
+      res.sendStatus(200)
+    } else {
+      res.status(401).send('Auth token invalid')
+    }
+  })
+
+  return router
+}
```

Comment:
Please move the router in a file called "api.ts". We try to keep the structure consistend across modules